rust/src/operations/vacuum.rs

```diff
@@ -214,18 +244,102 @@ impl std::future::IntoFuture for VacuumBuilder {
 struct VacuumPlan {
     /// What files are to be deleted
     pub files_to_delete: Vec<Path>,
+    /// If retention check is enabled
+    pub retention_check_enabled: bool,
+    /// Default retention in milliseconds
+    pub default_retention_millis: i64,
+    /// Overrided retention in milliseconds
+    pub specified_retention_millis: Option<i64>,
 }
 
 impl VacuumPlan {
     /// Execute the vacuum plan and delete files from underlying storage
-    pub async fn execute(self, store: &DeltaObjectStore) -> Result<VacuumMetrics, DeltaTableError> {
+    pub async fn execute(
+        self,
+        store: &DeltaObjectStore,
+        snapshot: &DeltaTableState,
+    ) -> Result<VacuumMetrics, DeltaTableError> {
         if self.files_to_delete.is_empty() {
             return Ok(VacuumMetrics {
                 dry_run: false,
                 files_deleted: Vec::new(),
             });
         }
 
+        let start_operation = DeltaOperation::VacuumStart {
+            retention_check_enabled: self.retention_check_enabled,
+            specified_retention_millis: self.specified_retention_millis,
+            default_retention_millis: self.default_retention_millis,
+        };
+
+        let end_operation = DeltaOperation::VacuumEnd {
+            status: String::from("COMPLETED"),
+        };
+
+        // let mut total_size = 0;
```

Comment:
I think if you want to collect this information, we should keep around the [ObjectMeta](https://docs.rs/object_store/latest/object_store/struct.ObjectMeta.html) here:

https://github.com/delta-io/delta-rs/blob/4226983f595cff004e905c3fde3c933aaae848d0/rust/src/operations/vacuum.rs#L203

instead of just getting the path. I think that should give you all the info you need, right?

Reply:
I'll take a look tonight! Thanks 💯

Reply:
I updated it now, one thing I noticed before is that we don't vacuum directories. I noticed it while vacuum a table with old partition folders. So I just keep the number of vacuumdirectories at 0.